### PR TITLE
fixes #2656 error connecting to vsphere after upgrade to ruby 1.9.3

### PR DESCRIPTION
**Pull request by abenari**

### CODE REVIEW

**Comment by domcleal** [comment]
Could you add a rake task too?  The trouble with a migration is it'll only run once, so might not be at the right time for some users.

**Comment by abenari** [comment]
@domcleal added a rake task.

**Comment by domcleal** [comment]
I'm unsure about the new change to the VMware model, it makes sense that it avoids user interaction (which is great), but is this token a security measure so we know we're always talking to the same host?  What's the point in storing it if we always allow it to change?

**Comment by abenari** [comment]
It's not always changing.
It is changing only if there is no stored hash, once the hash is stored an exception is raised on a mismatch.
In this patch since the hash stored in the db are obsolete, we delete them on migration and on first connection we will store the new value.
The other option I see is not to include the model change in the patch and ask our users to manually "test-connection" on every vmware compute-resource. Maybe  also show the new hash in the UI.

File: lib/tasks/reset_vmware.rake [hunk 1]
```diff
@@ -0,0 +1,8 @@
+namespace :vmware do
+  desc 'Reset vmware compute resource public keys in the database'
+  task :reset => :environment do
+    User.unscoped.as :admin do
+      Foreman::Model::Vmware.all.map{|cr| cr.attrs[:pubkey_hash] = "" ;cr.save}
```
Comment (ohadlevy) [inline lib/tasks/reset_vmware.rake:5]: instead of duplicating? maybe move it to a class method on the Vmware model?

**Pull request closed without merging**